src/nav/navReducer.js

```diff
@@ -64,6 +64,11 @@ const rehydrate = (state, action) => {
     return getStateForRoute('loading');
   }
 
+  // Dont switch to main UI if sharing screen is on.
+  if (state.routes.find(route => route.routeName === 'sharing')) {
+    return state;
+  }
+
```

Comment:
In the chat thread you started for this question, I think I figured out the mystery of what's happening here (and why notifications don't have the same trouble), and [found](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/need.20for.20conditional.20in.20navReducer's.20.60rehydrate.60.20for.20Sharing/near/945368) a cleaner solution. (And more complete! This one still leaves it the case that the sharing UI will briefly try to render in a situation where the state it has is bogus.)